akka-actor/src/main/scala/akka/util/Reflect.scala

```diff
@@ -43,7 +43,7 @@ private[akka] object Reflect {
    * @return a new instance from the default constructor of the given class
    */
   private[akka] def instantiate[T](clazz: Class[T]): T =
-    try clazz.newInstance
+    try clazz.getDeclaredConstructor().newInstance()
```

Comment:
this might have a performance implication for `.actorOf(Props[SomeActor])`.
Try with `ActorCreationBenchmark` and `ActorCreationPerfSpec`.

Maybe best to do these `newInstance` changes in a separate PR if the build changes are more urgent to get merged.

Reply:
will split them out